docs/sql/json.md

```diff
@@ -0,0 +1,243 @@
+# Dynamically-typed values and JSON support
+
+## The `VARIANT` type
+
+Values of `VARIANT` type are dynamically-typed.
+Any such value holds at runtime two pieces of information:
+- the data type
+- the data value
+
+Values of `VARIANT` type can be created by casting any other value to a `VARIANT`: e.g.
```

Comment:
so they can't be part of the type for a field in a table? but they can appear in views?

Reply:
The compiler places no restriction, this is mostly a question of adapters and serde really.